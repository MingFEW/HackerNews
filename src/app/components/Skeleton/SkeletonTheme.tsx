import React from 'react'
import { SkeletonTheme as RSkeletonTheme } from 'react-loading-skeleton'

interface SkeletonThemeProps {
  children: React.ReactNode
  baseColor?: string
  highLightColor?: string
}

export const SkeletonTheme: React.FC<SkeletonThemeProps> = props => {
  const {
    children,
    baseColor = 'var(--color-skeleton-base)',
    highLightColor = 'var(--color-skeleton-highlight)'
  } = props
  return (
    <RSkeletonTheme baseColor={baseColor} highlightColor={highLightColor}>
      {children}
    </RSkeletonTheme>
  )
}
