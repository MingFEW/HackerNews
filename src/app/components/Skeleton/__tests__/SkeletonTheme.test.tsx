import Skeleton from 'react-loading-skeleton'
import { createRenderer } from 'react-test-renderer/shallow'

import { SkeletonTheme } from '../SkeletonTheme'

const shallowRenderer = createRenderer()

describe('<SkeletonTheme />', () => {
  it('should render and match the snapshot', () => {
    shallowRenderer.render(
      <SkeletonTheme>
        <Skeleton count={3} />
      </SkeletonTheme>
    )
    const renderedOutput = shallowRenderer.getRenderOutput()
    expect(renderedOutput).toMatchSnapshot()
  })

  it('should match snapshot when props changed', () => {
    shallowRenderer.render(
      <SkeletonTheme baseColor="#fff" highLightColor="#333">
        <Skeleton count={3} />
      </SkeletonTheme>
    )
    const renderedOutput = shallowRenderer.getRenderOutput()
    expect(renderedOutput).toMatchSnapshot()
  })
})
