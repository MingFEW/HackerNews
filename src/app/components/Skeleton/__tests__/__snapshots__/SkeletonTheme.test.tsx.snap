// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SkeletonTheme /> should match snapshot when props changed 1`] = `
<SkeletonTheme
  baseColor="#fff"
  highlightColor="#333"
>
  <Skeleton
    count={3}
  />
</SkeletonTheme>
`;

exports[`<SkeletonTheme /> should render and match the snapshot 1`] = `
<SkeletonTheme
  baseColor="var(--color-skeleton-base)"
  highlightColor="var(--color-skeleton-highlight)"
>
  <Skeleton
    count={3}
  />
</SkeletonTheme>
`;
